metis-dereference/metis-dereference-rest/src/main/java/eu/europeana/metis/dereference/rest/config/ApplicationConfiguration.java

```diff
@@ -33,59 +35,57 @@
 @EnableScheduling
 @ComponentScan(basePackages = {"eu.europeana.metis.dereference.rest.controller",
     "eu.europeana.metis.dereference.rest.exceptions"})
-@PropertySource("classpath:dereferencing.properties")
 @EnableWebMvc
-public class ApplicationConfiguration implements WebMvcConfigurer, InitializingBean {
-
-  //Socks proxy
-  @Value("${socks.proxy.enabled}")
-  private boolean socksProxyEnabled;
-  @Value("${socks.proxy.host}")
-  private String socksProxyHost;
-  @Value("${socks.proxy.port}")
-  private String socksProxyPort;
-  @Value("${socks.proxy.username}")
-  private String socksProxyUsername;
-  @Value("${socks.proxy.password}")
-  private String socksProxyPassword;
-
-  //Mongo
-  @Value("${mongo.hosts}")
-  private String[] mongoHosts;
-  @Value("${mongo.port}")
-  private int mongoPort;
-  @Value("${mongo.username}")
-  private String mongoUsername;
-  @Value("${mongo.password}")
-  private String mongoPassword;
-  @Value("${mongo.application.name}")
-  private String mongoApplicationName;
-  @Value("${entity.db}")
-  private String entityDb;
-  @Value("${vocabulary.db}")
-  private String vocabularyDb;
-
-  //Valid directories list
-  @Value("${allowed.url.domains}")
-  private String[] allowedUrlDomains;
-  private MongoClient mongoClientEntity;
-  private MongoClient mongoClientVocabulary;
+public class ApplicationConfiguration implements WebMvcConfigurer {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ApplicationConfiguration.class);
+
+  private final ConfigurationPropertiesHolder propertiesHolder;
+  private final MongoClient mongoClientEntity;
+  private final MongoClient mongoClientVocabulary;
 
   /**
-   * Used for overwriting properties if cloud foundry environment is used
+   * Autowired constructor for Spring Configuration class.
+   *
+   * @param propertiesHolder the object that holds all boot configuration values
+   * @throws TrustStoreConfigurationException if the configuration of the truststore failed
    */
-  @Override
-  public void afterPropertiesSet() {
-    if (socksProxyEnabled) {
-      new SocksProxy(socksProxyHost, socksProxyPort, socksProxyUsername, socksProxyPassword).init();
+  public ApplicationConfiguration(ConfigurationPropertiesHolder propertiesHolder) throws TrustStoreConfigurationException {
+    mongoClientEntity = ApplicationConfiguration.initializeApplication(propertiesHolder);
+    mongoClientVocabulary = ApplicationConfiguration.initializeApplication(propertiesHolder);
```

Comment:
Two different instances of "the same" MongoClient, is each instances used for something different?

Reply:
One is for the entities and the other one for the vocabularies.
I left it as such since it was there.
But it seems reasonable to fix now so I've reduced it to one client.